src/main/java/services/AccountServiceOnHibernate.java

```diff
@@ -0,0 +1,114 @@
+package services;
+
+import org.hibernate.HibernateException;
+import org.hibernate.Session;
+import org.hibernate.SessionFactory;
+import org.hibernate.boot.registry.StandardServiceRegistryBuilder;
+import org.hibernate.cfg.Configuration;
+import org.hibernate.service.ServiceRegistry;
+import main.UserProfile;
+
+import java.util.List;
+
+/**
+ * Created by morev on 06.04.16.
+ */
+public class AccountServiceOnHibernate implements AccountService{
+    private final SessionFactory sessionFactory;
+
+    public AccountServiceOnHibernate() {
+        Configuration configuration = new Configuration();
+
+        configuration.setProperty("hibernate.dialect", "org.hibernate.dialect.MySQLDialect");
+        configuration.setProperty("hibernate.connection.driver_class", "com.mysql.jdbc.Driver");
+        configuration.setProperty("hibernate.connection.url", "jdbc:mysql://localhost:3306/BloodyDefenceDB");
+        configuration.setProperty("hibernate.connection.username", "bloodydefender");
+        configuration.setProperty("hibernate.connection.password", "stargate");
+        configuration.setProperty("hibernate.show_sql", "true");
+        configuration.setProperty("hibernate.hbm2ddl.auto", "create-drop");
+
+        configuration.addAnnotatedClass(UserProfile.class);
+
+        StandardServiceRegistryBuilder builder = new StandardServiceRegistryBuilder();
+        builder.applySettings(configuration.getProperties());
+        ServiceRegistry serviceRegistry = builder.build();
+        sessionFactory = configuration.buildSessionFactory(serviceRegistry);
+    }
+
+    @Override
+    public UserProfile getUser(long userID) {
+        try (Session session = sessionFactory.openSession()) {
```

Comment:
Любая работа с базой должна выполняться в блоке аналогичном следующему:
```
Session session;
try {
 session = sessionFactory.openSession();
    // Since we are in CMT, a JTA transaction would
    // already have been started.  This call essentially
    // no-ops
    session.getTransaction().begin();

    Number customerCount = (Number) session.createQuery( "select count(c) from Customer c" ).uniqueResult();

    // Since we did not start the transaction ( CMT ),
    // we also will not end it.  This call essentially
    // no-ops in terms of transaction handling.
    session.getTransaction().commit();
}
catch ( Exception e ) {
    LOGGER.error("Database operation failed", e)
    // again, the rollback call here would no-op (aside from
    // marking the underlying CMT transaction for rollback only).
    if ( session.getTransaction().getStatus() == TransactionStatus.ACTIVE
            || session.getTransaction().getStatus() == TransactionStatus.MARKED_ROLLBACK ) {
        session.getTransaction().rollback();
    }
    // handle the underlying error
}
finally {
    session.close();
}
```